bmc/boot_device.go

```diff
@@ -14,12 +14,29 @@ type BootDeviceSetter interface {
 	BootDeviceSet(ctx context.Context, bootDevice string, setPersistent, efiBoot bool) (ok bool, err error)
 }
 
+// BootDeviceOverrideGetter gets boot override settings for a machine
+type BootDeviceOverrideGetter interface {
+	BootDeviceOverrideGet(ctx context.Context) (override *BootDeviceOverride, err error)
+}
+
 // bootDeviceProviders is an internal struct to correlate an implementation/provider and its name
 type bootDeviceProviders struct {
 	name             string
 	bootDeviceSetter BootDeviceSetter
 }
 
+// bootOverrideProvider is an internal struct to correlate an implementation/provider and its name
+type bootOverrideProvider struct {
+	name          string
+	bootOverrider BootDeviceOverrideGetter
+}
+
+type BootDeviceOverride struct {
+	IsPersistent bool
+	IsEFIBoot    bool
+	Device       string
```

Comment:
What can be expected to be returned as `Device` here - does that differ per vendor?

Reply:
I would definitely prefer to have this be an enum or similar. However, `SetBootDevice` takes the boot device as a string, and I thought this information should match that interface.

```
func (c *Client) SetBootDevice(ctx context.Context, bootDevice string, setPersistent, efiBoot bool) (ok bool, err error) {
```

Reply:
I'd like to use constants where we can going ahead (for less ambiguity). 
Lets update the new method to accept a constant, and create an issue/PR to update `SetBootDevice`.

Reply:
Done. Mind having another look?